libs/timebar/src/chartsNew/track-graph.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useContext, useMemo } from 'react'
+import { area, curveStepAfter } from 'd3-shape'
+import { quantile } from 'simple-statistics'
+import ImmediateContext from '../immediateContext'
+import TimelineContext, { TimelineScale } from '../timelineContext'
+import { DEFAULT_CSS_TRANSITION } from '../constants'
+import { useFilteredChartData } from './common/hooks'
+import { getTrackY } from './common/utils'
+import { TimebarChartData, TimebarChartDataChunk, TimebarChartDataItem } from '.'
+
+const getMaxValues = (data: TimebarChartData) => {
+  const maxValues = data.map((trackGraphData: TimebarChartDataItem) => {
+    if (!trackGraphData.chunks.length) return 0
+    const itemValues = trackGraphData.chunks.reduce(
+      (acc: number[], currentChunk: TimebarChartDataChunk) => {
+        const chunkValues = currentChunk.values!.map((v) => v.value as number)
+        return acc.concat(chunkValues)
+      },
+      []
+    )
+
+    // https://online.stat.psu.edu/stat200/lesson/3/3.2
+    const q1 = quantile(itemValues, 0.25)
+    const q3 = quantile(itemValues, 0.75)
+    const iqr = Math.abs(q3 - q1)
+    const upperFence = Math.abs(q3 + iqr * 1.5)
+    const lowerFence = Math.abs(q1 - iqr * 1.5)
+
+    return upperFence > lowerFence ? upperFence : lowerFence
+  })
+  return maxValues
+}
+
+const getPaths = (
+  trackGraphData: TimebarChartDataItem,
+  numTracks: number,
+  trackIndex: number,
+  graphHeight: number,
+  overallScale: TimelineScale,
+  maxValue: number,
+  orientation: string
```

Comment:
I'd type the possible orientation options

Reply:
https://github.com/GlobalFishingWatch/frontend/pull/1213/commits/ff8177bc5b1f43fad2cf9e853b538e0ddd66fe7a